code/game/machinery/kitchen/smartfridge.dm

```diff
@@ -128,194 +126,253 @@
+		var/list/item_list = item_quants[item_index]
+		var/count = length(item_list)
+		if(count < 1)
+			continue
 
-/obj/structure/machinery/smartfridge/Topic(href, href_list)
-	if (..()) return 0
+		var/obj/example = item_list[1]
+		var/item_name = example.name
+		var/item_path = example.type
+		var/item_desc = example.desc
 
-	var/mob/user = usr
-	var/datum/nanoui/ui = nanomanager.get_open_ui(user, src, "main")
+		var/imgid = replacetext(replacetext("[item_path]", "/obj/item/", ""), "/", "-")
 
-	src.add_fingerprint(user)
+		var/item_type = "other"
+		if(ispath(item_path, /obj/item/reagent_container/glass/bottle/))
+			item_type = "bottle"
+		else if(ispath(item_path, /obj/item/storage/pill_bottle/))
+			item_type = "pills"
+
+		if (count > 0)
+			var/list/local_item = list(
+				"display_name" = capitalize(item_name),
+				"item" = item_path,
+				"index" = i,
+				"quantity" = count,
+				"image" = imgid,
+				"category" = item_type,
+				"desc" = item_desc,
+			)
+			local_items += list(local_item)
+
+	var/list/networked_items = list()
+	if(is_in_network())
+		for (var/i=1 to length(chemical_data.shared_item_storage))
+			var/item_index = chemical_data.shared_item_storage[i]
+			var/list/item_list = chemical_data.shared_item_storage[item_index]
+			var/count = length(item_list)
+			if(count < 1)
+				continue
+			var/obj/example = item_list[1]
+			var/item_name = example.name
+			var/item_path = example.type
+			var/item_desc = example.desc
+
+			var/imgid = replacetext(replacetext("[item_path]", "/obj/item/", ""), "/", "-")
+
+			var/item_type = "other"
+			if(ispath(item_path, /obj/item/reagent_container/glass/bottle/))
+				item_type = "bottle"
+			else if(ispath(item_path, /obj/item/storage/pill_bottle/))
+				item_type = "pills"
+
+			if (count > 0)
+				var/list/networked_item = list(
+					"display_name" = capitalize(item_name),
+					"item" = item_path,
+					"index" = i,
+					"quantity" = count,
+					"image" = imgid,
+					"category" = item_type,
+					"desc" = item_desc,
+				)
+				networked_items += list(networked_item)
+
+	.["storage"] = list(
+		"contents" = local_items,
+		"networked" = networked_items,
+	)
+
+/obj/structure/machinery/smartfridge/ui_assets(mob/user)
+	return list(get_asset_datum(/datum/asset/spritesheet/vending_products))
 
-	if (href_list["close"])
-		user.unset_interaction()
-		ui.close()
-		return FALSE
+/obj/structure/machinery/smartfridge/ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
+	. = ..()
+	if(.)
+		return
+	var/mob/user = usr
+	src.add_fingerprint(user)
 
-	if (href_list["toggletransfer"])
-		if(is_secure_fridge)
-			if(locked == FRIDGE_LOCK_COMPLETE)
-				to_chat(usr, SPAN_DANGER("Access denied."))
+	switch(action)
+		if("vend")
+			if(!ispowered)
+				to_chat(user, SPAN_WARNING("[src] has no power."))
 				return FALSE
-			if(!allowed(usr) && locked == FRIDGE_LOCK_ID)
-				to_chat(usr, SPAN_DANGER("Access denied."))
+			if (!in_range(src, usr))
+				return FALSE
```

Comment:
should be handled by ui state, no?

Reply:
Was just copied out of original, stripped.